File: java_version/src/main/java/com/bruce3x/kotlin/demo/MainActivity.java
package com.bruce3x.kotlin.demo;

import android.os.Bundle;
import android.view.View;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private FoodOrderService service = new FoodOrderService(false);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();
    }

    private void initViews() {
        RadioGroup options = findViewById(R.id.rg_options);
        RadioButton foodA = findViewById(R.id.rb_food_a);
        RadioButton foodB = findViewById(R.id.rb_food_b);
        RadioButton foodC = findViewById(R.id.rb_food_c);
        RadioButton foodD = findViewById(R.id.rb_food_d);
        foodA.setText(FoodOrderService.FOOD_A);
        foodB.setText(FoodOrderService.FOOD_B);
        foodC.setText(FoodOrderService.FOOD_C);
        foodD.setText("山珍海味");

        View btnSelect = findViewById(R.id.btn_select);
        View btnReset = findViewById(R.id.btn_reset);

        TextView tvDetail = findViewById(R.id.tv_detail);
        tvDetail.setText(service.getOrderDetail());

        btnSelect.setOnClickListener(v -> {
            // 获取选中的菜名
            RadioButton selected = findViewById(options.getCheckedRadioButtonId());
            String food = selected.getText().toString();

            try {
                service.select(food);
            } catch (FoodOrderService.FoodUnavailableException e) {
                Toast.makeText(MainActivity.this, e.getMessage(), Toast.LENGTH_SHORT).show();
            } finally {
                tvDetail.setText(service.getOrderDetail());
            }
        });

        btnReset.setOnClickListener(v -> {
            service.reset();
            tvDetail.setText(service.getOrderDetail());
        });
    }
}

File: kotlin_version/src/main/java/com/bruce3x/kotlin/demo/MainActivity.kt
package com.bruce3x.kotlin.demo

import android.os.Bundle
import android.view.View
import android.widget.RadioButton
import android.widget.RadioGroup
import android.widget.TextView
import android.widget.Toast

import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private val service = FoodOrderService(false)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()
    }

    private fun initViews() {
        val options: RadioGroup = findViewById(R.id.rg_options)
        val foodA: RadioButton = findViewById(R.id.rb_food_a)
        val foodB: RadioButton = findViewById(R.id.rb_food_b)
        val foodC: RadioButton = findViewById(R.id.rb_food_c)
        val foodD: RadioButton = findViewById(R.id.rb_food_d)
        foodA.text = FoodOrderService.FOOD_A
        foodB.text = FoodOrderService.FOOD_B
        foodC.text = FoodOrderService.FOOD_C
        foodD.text = "山珍海味"

        val btnSelect = findViewById<View>(R.id.btn_select)
        val btnReset = findViewById<View>(R.id.btn_reset)

        val tvDetail = findViewById<TextView>(R.id.tv_detail)
        tvDetail.text = service.getOrderDetail()

        btnSelect.setOnClickListener {
            // 获取选中的菜名
            val selected: RadioButton = findViewById(options.checkedRadioButtonId)
            val food = selected.text.toString()

            try {
                service.select(food)
            } catch (e: FoodOrderService.FoodUnavailableException) {
                Toast.makeText(this@MainActivity, e.message, Toast.LENGTH_SHORT).show()
            } finally {
                tvDetail.text = service.getOrderDetail()
            }
        }

        btnReset.setOnClickListener {
            service.reset()
            tvDetail.text = service.getOrderDetail()
        }
    }
}